packages/features/users/components/UserTable/BulkActions/EventTypesList.tsx

```diff
@@ -0,0 +1,262 @@
+
+                    const ids = events.map((event) => event.id);
+                    const areAllUsersHostForTeam = selectedUsers.every((user) =>
+                      events.every((event) => event.hosts.some((host) => host.userId === user.id))
+                    );
+                    const isSelected = ids.every(
+                      (id) =>
+                        selectedEvents.has(id) || (areAllUsersHostForTeam && !removeHostFromEvents.has(id))
+                    );
+                    return (
+                      <>
+                        <ListItem
+                          isTeam
+                          onSelect={() => {
+                            if (!isSelected) {
+                              // Add current team and its event
+                              addValue(selectedTeams, setSelectedTeams, [teamId]);
+                              addValue(selectedEvents, setSelectedEvents, ids);
+                              setRemoveHostFromEvents(new Set());
+                            } else {
+                              const eventIdsWhereAllUsersAreHosts = events
+                                .filter((event) =>
+                                  selectedUsers.every((user) =>
+                                    event.hosts.some((host) => host.userId === user.id)
+                                  )
+                                )
+                                .map((event) => event.id);
+
+                              addValue(
+                                removeHostFromEvents,
+                                setRemoveHostFromEvents,
+                                eventIdsWhereAllUsersAreHosts
+                              );
+                              // Remove selected team and its event
+                              removeValue(selectedEvents, setSelectedEvents, ids);
+                              removeValue(selectedTeams, setSelectedTeams, [teamId]);
+                            }
+                          }}
+                          isSelected={isSelected}
+                          text={team.profile.name || ""}
+                          key={team.profile.name}
+                        />
+                        {events.map((event) => {
+                          const hosts = event.hosts;
+                          const areAllUsersHostForEventType = selectedUsers.every((user) =>
+                            hosts.some((host) => host.userId === user.id)
+                          );
+                          const isSelected =
+                            selectedEvents.has(event.id) ||
+                            (areAllUsersHostForEventType && !removeHostFromEvents.has(event.id));
+                          return (
+                            <ListItem
+                              isTeam={false}
+                              onSelect={() => {
+                                if (!isSelected) {
+                                  if (areAllUsersHostForEventType) {
```

Comment:
If there are 3 events A, B, C and user is part of event B. If you click on team, all events gets selected. If you uncheck team, all events get unchecked. Mind you we have not clicked apply yet so user is still a host of event B, but now if you click Event B nothing happens (it should check event b).

Maybe I could name this better lmk what you think